casr/src/util.rs

```diff
@@ -248,3 +251,43 @@ pub fn log_progress(processed_items: &RwLock<usize>, total: usize) {
         std::thread::sleep(std::time::Duration::from_millis(1000));
     }
 }
+
+/// Get output of target command with specified timeout
+///
+/// # Arguments
+///
+/// * `command` - target command with args
+///
+/// * `timeout` - target command timeout (in seconds)
+///
+/// * `error_on_timeout` - throw an error if timeout happens
+///
+/// # Return value
+///
+/// Command output
+pub fn get_output(command: &mut Command, timeout: u64, error_on_timeout: bool) -> Result<Output> {
+    // If timeout is specified, spawn and check timeout
+    // Else get output
+    if timeout != 0 {
+        let mut child = command
+            .spawn()
+            .with_context(|| "Failed to start command: {command:?}")?;
+        if child
+            .wait_timeout(Duration::from_secs(timeout))
+            .unwrap()
+            .is_none()
+        {
+            let _ = child.kill();
+            if error_on_timeout {
+                error!("Timeout: {:?}", command);
```

Comment:
Just use error instead of warn, remove this parameter

Reply:
Replaced with bail